vignettes/decorate-module-output.Rmd

```diff
@@ -0,0 +1,467 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` is a powerful `shiny`-based framework with built-in modules for interactive data analysis. 
+This document outlines the customization options available for modifying the output of `teal` modules. 
+You will learn how to use [`teal_transform_module`](https://insightsengineering.github.io/teal/latest-tag/reference/teal_transform_module.html) to modify and enhance the objects created by `teal::modules()`, 
+enabling you to tailor the outputs to your specific requirements without rewriting the original module code.
+
+## Decorators
+
+One of ways of adjusting input data or customizing module outputs in `teal` is the usage of `transformators` 
+created through `teal_transform_module`. This document will focus on modifying outputs, and for clarity, we will refer 
+to these output-modifying objects as decoration objects or decorators.
+
+In below chapter we will present how to create the simplest static decorator with just a server part. Later, we will
+present examples on more advanced usage, where decorators contain UI. You will also learn about a convenience 
+function that makes it easier to write decorators, called [`make_teal_transform_server`](https://insightsengineering.github.io/teal/latest-tagx/reference/make_teal_transform_server.html). The chapter ends with an 
+example module that utilizes decorators and a snippet that uses this module in `teal` application.
+
+### Server
+
+The simplest way to create a decorator is to use `teal_transform_module` without specifying a `ui`.
+This approach adds functionality solely to the server code of the module. 
+In the following example, we assume that the module contains an object (of class `ggplot2`) named `plot`. 
+We modify the title and x-axis label of plot:
+
+```{r}
+library(teal)
+static_decorator <- teal_transform_module(
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive({
+        req(data())
+        within(data(), {
+          plot <- plot +
+            ggtitle("This is title") +
+            xlab("x axis")
+        })
+      })
+    })
+  }
+)
+```
+
+To simplify the repetitive elements of writing new decorators 
+(e.g., `function(id, data), moduleServer, reactive, within(data, ...)`), 
+you can use the `make_teal_transform_server(expr)` convenience function, which takes a `language` as input:
```

Comment:
every time we introduce a new function - let's link it
```suggestion
you can use the [`make_teal_transform_server()`](https://insightsengineering.github.io/teal/latest-tagx/reference/make_teal_transform_server.html) convenience function, which takes a `language` as input:
```

Reply:
Sure, it was linked a couple sentences before, but this does not hurt